MapboxCoreNavigation/RouteController.swift

```diff
@@ -154,13 +154,26 @@ extension RouteController: CLLocationManagerDelegate {
         lastUserDistanceToStartOfRoute = Double.infinity
     }
     
-    public func userIsOnRoute(_ location: CLLocation) -> Bool {
-        // Find future location of user
+    
+    /**
+     Returns the calculated reroute distance.
+     */
+    public func getRerouteRadius(_ horizontalAccuracy: CLLocationAccuracy) -> CLLocationDistance {
+        return min(RouteControllerMaximumDistanceBeforeRecalculating, horizontalAccuracy + RouteControllerUserLocationSnappingDistance)
+    }
+    
+    /**
+     Takes a `CLLocation` and returns a `CLLocation` infront of the provided location given the current speed.
+     */
+    public func getDeadReckoningLocation(_ location: CLLocation) -> CLLocation {
         let metersInFrontOfUser = location.speed * RouteControllerDeadReckoningTimeInterval
         let locationInfrontOfUser = location.coordinate.coordinate(at: metersInFrontOfUser, facing: location.course)
-        let newLocation = CLLocation(latitude: locationInfrontOfUser.latitude, longitude: locationInfrontOfUser.longitude)
-        let radius = min(RouteControllerMaximumDistanceBeforeRecalculating,
-                         location.horizontalAccuracy + RouteControllerUserLocationSnappingDistance)
+        return CLLocation(latitude: locationInfrontOfUser.latitude, longitude: locationInfrontOfUser.longitude)
+    }
+    
+    public func userIsOnRoute(_ location: CLLocation) -> Bool {
```

Comment:
One way to tell that the method’s name needs revision is that the first argument is called `location` internally (and is a CLLocation) but is called a “route” externally; a location isn’t a route. To fix this issue, move this method to an extension on CLLocation and make its only argument a Route. There’s already a similar extension method in Geometry.swift.